src/Java.Interop/Java.Interop/JniBuiltinMarshalers.cs

```diff
@@ -233,12 +234,12 @@ public override Boolean CreateGenericValue (ref JniObjectReference reference, Jn
 			return JniBoolean.GetValueFromJni (ref reference, options, targetType);
 		}
 
-		public override JniValueMarshalerState CreateGenericArgumentState (Boolean value, ParameterAttributes synchronize = ParameterAttributes.In)
+		public override JniValueMarshalerState CreateGenericArgumentState ([MaybeNull]Boolean value, ParameterAttributes synchronize = ParameterAttributes.In)
```

Comment:
Do we have commit message info for why the introduction of `[MaybeNull]`?  Why wasn't this needed before?

Additionally, there should be a corresponding change to [`src/Java.Interop/Java.Interop/JniBuiltinMarshalers.tt`](https://github.com/xamarin/java.interop/blob/master/src/Java.Interop/Java.Interop/JniBuiltinMarshalers.tt).  Visual Studio (for Mac?) should update `JniBuiltinMarshalers.cs` based on `JniBuiltinMarshalers.tt`.

Reply:
Probably "too late", but did we need the `Java.Interop.dll` changes intermixed with the `generator` changes?

Reply:
The method it is overriding specifies `[MaybeNull]`, so you get a "nullability mismatch warning":

https://github.com/xamarin/java.interop/blob/master/src/Java.Interop/Java.Interop/JniValueMarshaler.cs#L221

I can't find an instance of it right now, from what I can tell this warning was still be tweaked by the Roslyn team, and it only appears on newer versions of the C# compiler.

Reply:
I removed these changes for now, as they'll need to be done in the `.tt` files.  There's about 60 warnings that I am seeing now:

https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3665517&view=logs&j=f31c9f97-4411-58e7-49ac-fc73f645e6b6&t=cbeb8522-7b64-5cc0-7bee-eff0e164a409

I don't know how I missed these before.  I suggest we go ahead and continue getting this PR merged and then we can go back and clean up those warnings.